### PR TITLE
frontend: add typography component

### PR DESCRIPTION
**Pull request by dschaller**
### Description

Add a new Typography component based on the figma designs.

### Testing Performed

manual and storybook

### CODE REVIEW

**Review by mikecutalo — APPROVED** [review]
Played around with it a bit in storybook and looks good!

File: frontend/packages/core/src/typography.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    | "body3"
+    | "caption1"
+    | "caption2"
+    | "overline"
+    | "input";
+  children: React.ReactNode;
+}
+
+const Typography = ({ variant, children }: TypographyProps) => {
+  return <StyledTypography data-variant={variant}>{children}</StyledTypography>;
```
Comment (danielhochman) [inline frontend/packages/core/src/typography.tsx:137]: does this need to be a data prop?

Comment (dschaller) [reply frontend/packages/core/src/typography.tsx:137]: yes, div doesn't have a variant prop

Comment (danielhochman) [reply frontend/packages/core/src/typography.tsx:137]: This seems to work fine.

```typescript
interface MyDivProps {
  colorz: string;
}

const MyStyledDiv = styled.div<MyDivProps>(
  {}, props => ({color: props.colorz})
);

const MyDiv = () => <MyStyledDiv colorz="blue" />
```

Comment (danielhochman) [reply frontend/packages/core/src/typography.tsx:137]: or just
```typescript
const MyStyledDiv = styled.div<{colorz: string}>(
  {}, props => ({color: props.colorz})
);

const MyDiv = () => <MyStyledDiv colorz="blue" />
```

Comment (dschaller) [reply frontend/packages/core/src/typography.tsx:137]: Oh yeah, I guess I can define the interface for it. I was relying on the existing props

**Pull request merged**